### ISSUE TITLE
Bug: Loading spinner missing in lower grid of Emissions screen

### ISSUE THREAD

**Issue by esaber76** [US-EPA-CAMD/PI-1#6449]
There used to also be a loading spinner in the lower grid of the Emissions screen while data was still loading for a view.   As of now, the following is displayed to a user while the data is loading. 

![Image](https://github.com/user-attachments/assets/af55805e-63e0-4ebe-b29e-06a6805a1cf7)

When user clicks the filter, "loading data" should appear until the data is ready to display.

**Comment by ibarra-michelle** [comment]
Hello @esaber76: I spoke internally with @j-tafoya, who helped lead the design of the Emissions module. We should revisit the expected workflow of what should happen here together. Since this ticket is marked as a low, I imagine we'll talk about this ticket next week after we sort through our critical and high tickets first, thank you. 

**Requirements**
The message that is currently displayed when first accessing the screen should change **_from_**, "There are no records to display," **_to_** "Select a Reporting Period, Location, and Template and "Apply Filter" to view the data"
 
_THEN_, after the user clicks the "Apply filter" button, the spinner appears. After the data is loaded (if there is data), the spinner should disappear. If there is no data, then a new message should state,"There are no records to display. Select another Reporting Period, Location, and Template, then click the Apply Filter button to view the data."

FYA: @bryanramirez1 @annalbrecht @spetros-do @alangmaid

**Comment by Jhernandez29** [comment]
Created new branch https://github.com/US-EPA-CAMD/easey-ecmps-ui/tree/fix/%236449-emission-Loading-spinner-missing and commit lasted code changes. The spinner still is not doing what the requirements are. Still needs work.

**Comment by Jhernandez29** [comment]
created PR https://github.com/US-EPA-CAMD/easey-ecmps-ui/pull/1511 and move ticket to peer review

**Comment by maxdiebold-erg** [comment]
_Tested in the Development environment._

**Facility:** Comanche (470)
**Configuration:** 1

- The initial instructions are shown when no filters are applied:

![Image](https://github.com/user-attachments/assets/5e4a5ef7-0597-4573-9fab-573384652265)

- The loading spinner is shown when options are selected and "Apply Filter" is clicked:

![Image](https://github.com/user-attachments/assets/592194f1-f92c-4fbb-b0b6-5e669163f992)

- This resolves to a table of records:

![Image](https://github.com/user-attachments/assets/360d0fbc-d6b2-4ccd-bd3e-d2beb72c6cba)

- When an option is changed, the initial message is shown again.

- If there is no data in a view, the message "There are no records to display" is shown briefly, before changing back to the initial instructions.

![Image](https://github.com/user-attachments/assets/2aa9dd06-6c75-498b-8fac-0765d6f1ee8f)

@ibarra-michelle @esaber76 The timeout on the "no records" message is currently set to 1 second. **Should this be increased? Alternatively, should it revert back to the initial message only when a selection is changed?**

**Comment by ibarra-michelle** [comment]
> [@ibarra-michelle](https://github.com/ibarra-michelle) [@esaber76](https://github.com/esaber76) The timeout on the "no records" message is currently set to 1 second. **Should this be increased? Alternatively, should it revert back to the initial message only when a selection is changed?**

@maxdiebold-erg - let me gather some additional info on your 1st question and I'll follow-up in a new comment. 

- For your 2nd question, do you mean to say if there is no data in a view, perhaps we should consider updating the the message to, "There are no records to display. Select another Reporting Period, Location, and Template, then click the Apply Filter button to view the data."?

- Also, I noticed we have a typo in the requirements for first message. Could you please update the text to, "Select another Reporting Period, Location, and Template, then click the Apply Filter button to view the data."

**Comment by ibarra-michelle** [comment]
> [@ibarra-michelle](https://github.com/ibarra-michelle) [@esaber76](https://github.com/esaber76) The timeout on the "no records" message is currently set to 1 second. **Should this be increased? Alternatively, should it revert back to the initial message only when a selection is changed?**

Proposed response for question 1:
1. @maheese and @esaber76 I think we should increase the timeout. Is there any perf data indicating how long it is taking for the data to load? For example, is a 5-minute timeout enough or too little/much time? Thoughts on the timeout length?

FYA @bryanramirez1 @j-tafoya @chrisgitsit

**Comment by maxdiebold-erg** [comment]
@ibarra-michelle Sorry let me rephrase the questions: the spinner shows for as long as it takes to load the data. Then, if there is no data for the selected view, the message _"There are no records to display"_ shows for 1 second only before changing back to _"Select a Reporting Period, Location, and Template and Apply Filter to view the data"_. I'm asking if that is the desired behavior, if message should show for longer than 1 second, or if it should just show _"There are no records to display"_ (or _"There are no records to display. Select another Reporting Period, Location, and Template, then click the Apply Filter button to view the data."_) until an option is changed.

**Comment by esaber76** [comment]
@ibarra-michelle @maheese @bryanramirez1 @chrisgitsit @maxdiebold-erg (or "There are no records to display. Select another Reporting Period, Location, and Template, then click the Apply Filter button to view the data.") makes sense to me...

**Comment by ibarra-michelle** [comment]
> [@ibarra-michelle](https://github.com/ibarra-michelle) Sorry let me rephrase the questions: the spinner shows for as long as it takes to load the data. Then, if there is no data for the selected view, the message _"There are no records to display"_ shows for 1 second only before changing back to _"Select a Reporting Period, Location, and Template and Apply Filter to view the data"_. I'm asking if that is the desired behavior, if message should show for longer than 1 second, or if it should just show _"There are no records to display"_ (or _"There are no records to display. Select another Reporting Period, Location, and Template, then click the Apply Filter button to view the data."_) until an option is changed.

Thx, @maxdiebold-erg - let's move forward with the text, "There are no records to display. Select another Reporting Period, Location, and Template, then click the Apply Filter button to view the data" until an option is changed.

**Comment by ibarra-michelle** [comment]
> [@ibarra-michelle](https://github.com/ibarra-michelle) [@esaber76](https://github.com/esaber76) The timeout on the "no records" message is currently set to 1 second. **Should this be increased? Alternatively, should it revert back to the initial message only when a selection is changed?**

@maxdiebold-erg @Jhernandez29 , I spoke to @bryanramirez1 to confirm:
1.  The timeout on the "no records" message should not exist, it should be a static message. 
2.  The timeout on the "no records" message should be changed to, "There are no records to display. Select another Reporting Period, Location, and Template, then click the Apply Filter button to view the data." I will update the requirements text above.

Thanks for pointing this out and please let me know if you have additional questions.

FYA @esaber76 @maheese

**Comment by maxdiebold-erg** [comment]
The lower grid of the Emissions module now shows the corrected initial message:

![Image](https://github.com/user-attachments/assets/8ccfff39-3cb3-40c8-b775-010a813e22f0)

In the case of a view with no records, the updated "no records" message is shown, and remains until the options are changed:

![Image](https://github.com/user-attachments/assets/64ff58e5-6f7a-4ab4-bd33-456b3194baf6)

**Comment by chrisgitsit** [comment]
Confirmed using 2022Q4 ORIS 470 Unit 2
